### DISCUSSION TITLE
Nested <a> tag cause rehydration to insert node at wrong place

### DISCUSSION THREAD

**Discussion by armedi** [vercel/next.js#35063] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

  Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.2.0: Sun Nov 28 20:29:10 PST 2021; root:xnu-8019.61.5~1/RELEASE_ARM64_T8101
    Binaries:
      Node: 16.14.0
      npm: 8.3.1
      Yarn: 1.22.17
      pnpm: 6.11.0
    Relevant packages:
      next: 12.1.1-canary.6
      react: 17.0.2
      react-dom: 17.0.2

### What browser are you using? (if relevant)

Chrome

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

When `<a>` tag is nested, the hydration insert wrong element

This structure

```jsx
<main>
  <a href="#">
    <div style={{ background: 'aqua', border: '1px solid red' }}>
      <div style={{ height: '200px' }}>
        <a href="#">
          This box has background color AQUA
        </a>
      </div>
    </div>
  </a>
  <div style={{ background: 'orange', border: '1px solid red' }}>
    <div style={{ height: '200px' }}>This box should have background color ORANGE</div>
  </div>
</main>
```

after hydration becomes

```jsx
<main>
  <a href="#">
    <div style="background: aqua; border: 1px solid red">
      <div style="height: 200px">
        <a href="#">This box has background color AQUA</a>
      </div>
    </div>
  </a>
  <div style="background: aqua; border: 1px solid red">
    <div style="height: 200px">This box should have background color ORANGE</div>
  </div>
</main>

```

The last div should have style "background: orange;" but instead it has "background: aqua"

### Expected Behavior

The hydrated element is

```jsx
<main>
  <a href="#">
    <div style="background: aqua; border: 1px solid red">
      <div style="height: 200px">
        <a href="#">This box has background color AQUA</a>
      </div>
    </div>
  </a>
  <div style="background: orange; border: 1px solid red">
    <div style="height: 200px">This box should have background color ORANGE</div>
  </div>
</main>

```

### To Reproduce

https://github.com/armedi/next-nested-a

**Comment by balazsorban44** [comment]
Nested `a` elements are forbidden in both HTML4 and HTML5 and you should avoid them anyway.

HTML4

> Links and anchors defined by the A element must not be nested; an A element must not contain any other A elements.

https://www.w3.org/TR/html401/struct/links.html#h-12.2.2

HTML5
> The [a](https://www.w3.org/TR/2011/WD-html5-author-20110809/the-a-element.html#the-a-element) element may be wrapped around entire paragraphs, lists, tables, and so forth, even entire sections, so long as there is no interactive content within (e.g. buttons or other links).

https://www.w3.org/TR/2011/WD-html5-author-20110809/the-a-element.html

**Reply by armedi** [reply]
thanks @balazsorban44 

In our case we can't really avoid this structure for now, because the content is made by our users.

One workaround I've tried and seems to work is by wrapping the first `<a>` with `<div>`

```jsx
<main>
  <div>
    <a href="#">
      <div style="background: aqua; border: 1px solid red">
        <div style="height: 200px">
          <a href="#">This box has background color AQUA</a>
        </div>
      </div>
    </a>
  </div>
  <div style="background: aqua; border: 1px solid red">
    <div style="height: 200px">
      This box should have background color ORANGE
    </div>
  </div>
</main>
```
